Processing_Side_5.21

```diff
@@ -28,17 +28,41 @@ void setup() {
   rows = testImage.height;
   columns = testImage.width;
   
-  for(int i = (testImage.pixels.length - 1); i >= 0; i--) {
+  for (int r = rows; r > 0; r--) {
```

Comment:
It'll save you a little typing later if you use zero-based indexes for row and column offsets, because it simplifies the math a little.  For instance if you use `for (int r = rows - 1; r >= 0; r--)` here and `for (int c = columns - 1; c >= 0; c--)` down below, the pixelLocation assignments become simpler: `r * columns + c`.